src/classes/UTIL_OrgTelemetry_SObject_BATCH.cls

```diff
@@ -0,0 +1,119 @@
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2018
+* @group Utilities
+* @description To prevent a timeout error counting Opportunity records using specific filters, this batch job
+* is built to be generic enough to be executed in different configurations to count whatever large object requires
+* counting. Stateful is used to hold the count variables and an Enum is used to define the mode the batch job
+* is executing in.
+*/
+public class UTIL_OrgTelemetry_SObject_BATCH implements Database.Batchable<SObject>, Database.Stateful {
+
+    /**
+     * @description BatchJobMode is to add support for other query types in this same batch job
+     * that aren't yet needed, but may be in the future
+     */
+    public enum TelemetryBatchJobMode {
+        RECURRING_DONATIONS
+    }
+
+    /** @description The job mode this particular batch iteration is executing in */
+    private UTIL_OrgTelemetry_SObject_BATCH.TelemetryBatchJobMode jobMode;
+
+    /** @description Stateful counter vars used for the different counting jobs */
+    private Integer countAllRDOpps = 0;
+    private Integer countOpenRDOpps = 0;
+
+    private static final String recurringDonationOpenStatusValue = System.Label.npe03.RecurringDonationOpenStatus;
+
+
+    /**
+     * @description Constructor
+     */
+    public UTIL_OrgTelemetry_SObject_BATCH(UTIL_OrgTelemetry_SObject_BATCH.TelemetryBatchJobMode jobMode) {
+        this.jobMode = jobMode;
+    }
+
+    /**
+     * @description Execute the appropriate query to use in this batch job.
+     */
+    public Database.QueryLocator start(Database.BatchableContext context) {
+        switch on (jobMode) {
+            when RECURRING_DONATIONS {
+                return Database.getQueryLocator(
+                        [SELECT Id, npe03__Recurring_Donation__r.npe03__Open_Ended_Status__c
+                        FROM Opportunity
+                        WHERE npe03__Recurring_Donation__c != null]);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * @description For each batch iteration, loop through the records and count as needed into
+     * stateful vars
+     */
+    public void execute(Database.BatchableContext context, List<SObject> scope) {
+        for (SObject record : scope) {
+            switch on (jobMode) {
+                when RECURRING_DONATIONS {
+                    getRecurringDonationsCounts((Opportunity)record);
```

Comment:
Missing space, ie `((Opportunity) record);`

Reply:
I didn't think that a space is required when casting? I treat this something like `:recordId` because the character directly manipulates the var. Different than curly brackets because those are structure control characters.